### PR TITLE
[Tests-Only] Navigate to private link as an owner of the file, collaborator and an unauthorized user

### PR DESCRIPTION
**Pull request by Talank**
## Description
Test is added for navigating to a file using perma-link as different users like owner, collaborator and unauthorized person

## Related Issue
- Fixes https://github.com/owncloud/phoenix/issues/3096
- Found https://github.com/owncloud/phoenix/issues/3243

## How Has This Been Tested?
locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/webUIFiles/copyPrivateLinks.feature [hunk 1]
```diff
@@ -16,3 +16,4 @@ Feature: copy path as a permanent link
       | resource_name   |
       | "lorem.txt"     |
       | "simple-folder" |
+
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUIFiles/copyPrivateLinks.feature]: get rid of this extra line

File: tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature [hunk 2]
```diff
@@ -18,4 +18,22 @@ Feature: Access private link
...
     When user "user1" has logged in using the webUI
     Then the app-sidebar for file "lorem.txt" should be visible on the webUI
-    And the "versions" details panel should be visible
\ No newline at end of file
+    And the "versions" details panel should be visible
+
+  Scenario: Access private link as a collaborator
+    Given user "user2" has been created with default attributes
+    And user "user1" has shared file "lorem.txt" with user "user2" with "read" permissions
+    And the user navigates to the private link created by user "user1" for file "lorem.txt"
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature]: `And the user has navigated to the private link created by user "user1" for file "lorem.txt"`

File: tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature [hunk 3]
```diff
@@ -18,4 +18,22 @@ Feature: Access private link
...
+    When user "user2" has logged in using the webUI
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature]: `When user "user2" logs in using the webUI`

File: tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature [hunk 4]
```diff
@@ -18,4 +18,22 @@ Feature: Access private link
...
+    And user "user2" has logged in using the webUI
+    When the user navigates to the private link created by user "user1" for file "lorem.txt"
+    Then the app-sidebar for file "lorem (2).txt" should be visible on the webUI
+    And the "versions" details panel should be visible
+
+  Scenario: Access the private link as an user that does not have permissions to see the file
+    Given user "user2" has been created with default attributes
+    And user "user2" has logged in using the webUI
+    When the user navigates to the private link created by user "user1" for file "lorem.txt"
+    Then the user should see the following error message on the login card dialog
```
Comment (dpakach) [inline tests/acceptance/features/webUIPrivateLinks/accessingPrivateLinks.feature]: There should be a better error please raise issue for this and tag it or check if there is already an issue.

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 5]
```diff
@@ -58,6 +58,13 @@ Then('the error message {string} should be displayed on the webUI dialog prompt'
     .expect.element('@ocDialogPromptAlert').text.to.equal(message)
 })
 
+Then('the user should see the following error message on the login card dialog', function (message) {
+  return client
+    .page.publicLinkPasswordPage()
+    .waitForElementVisible('@loginCardDialogBox')
+    .expect.element('@loginCardDialogBox').text.to.equal(message)
```
Comment (dpakach) [inline tests/acceptance/stepDefinitions/generalContext.js]: It's better not use the selector names directly from the context files, rather create a method in the page object that returns the error message, but since we have other cases like this we can do that in different PR.

**Comment by swoichha** [comment]
LGTM +1

**Comment by swoichha** [comment]
LGTM
